sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -149,14 +289,14 @@ static ProcessElementMethod create(
 
     /** Whether this {@link DoFn} uses a Single Window. */
     public boolean usesSingleWindow() {
-      return extraParameters().contains(Parameter.BOUNDED_WINDOW);
+      return extraParameters().contains(Parameter.boundedWindow());
```

Comment:
I'd be more comfortable if this referred to the BOUNDED_WINDOW constant (and if boundedWindow() did not exist), since Parameter classes do not implement equals/hashCode and boundedWindow() does not make the promise that it'll always return the same object. Likewise for the other parameters.